client/src/components/ResultItem/ResultItem.js

```diff
@@ -1,13 +1,34 @@
+/* eslint-disable no-eval */
 import './ResultItem.css';
 
 import { Highlight } from 'react-instantsearch-dom';
 import PropTypes from 'prop-types';
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 
 /* Algolia's Hit: display a result item from search */
 function ResultItem(props) {
   // a hit (result) object that contains all info about a result item
   const { hit } = props;
+  const history = useHistory();
+
+  const handleAnalysis = () => {
+    const { data } = hit;
+    const ad = {
+      headline: data.headline,
+      headlineAnalysis: {
+        sentiment: JSON.parse(data.headlineSentiment),
+        // use eval() due to single-quote appearances in JSON
```

Comment:
Minor nit: please have whitespace between code and comment to improve readability.